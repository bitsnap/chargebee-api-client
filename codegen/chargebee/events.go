package chargebee

import (
	"github.com/bitsnap/chargebee-api-client/codegen/client"
)

// GenerateListEvents generates chargebee client code to fetch all events
//
// API: https://{site}.chargebee.com/api/v2/events
//
// Documentation: https://apidocs.chargebee.com/docs/api/events?lang=curl#list_events
func GenerateListEvents() string {
	return client.GenerateList("Events", "Event", "chargebee.com/api/v2/events")
}

// GenerateRetrieveEvent generates chargebee client code to retrieve specific event
//
// API: https://{site}.chargebee.com/api/v2/events/{event-id}
//
// Documentation: https://apidocs.chargebee.com/docs/api/events?lang=curl#retrieve_an_event
func GenerateRetrieveEvent() string {
	return client.GenerateRetrieve("Event", "chargebee.com/api/v2/events")
}

// GenerateEventsModel generates chargebee event domain model
//
// Documentation: https://apidocs.chargebee.com/docs/api/events?lang=curl#event_attributes
func GenerateEventsModel() string {
	return client.GenerateModel("Event", "Event attributes", "https://apidocs.chargebee.com/docs/api/events")
}
