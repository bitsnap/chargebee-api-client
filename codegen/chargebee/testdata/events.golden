# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee events code generation when generating Chargebee API client should be able to generate events models" = '''
(string) (len=555) "type Event struct {\n    Id string `json:\"id\" validate:\"required\"`\n    OccurredAt *time.Time `json:\"occurred_at\" validate:\"required\"`\n    Source enums.SourceEnum `json:\"source\" validate:\"required\"`\n    User string `json:\"user\"`\n    EventType enums.EventTypeEnum `json:\"event_type\"`\n    ApiVersion enums.ApiVersionEnum `json:\"api_version\"`\n    OriginUser string `json:\"origin_user\"`\n    Webhooks []Webhook `json:\"webhooks\"`\n    Id string `json:\"id\" validate:\"required\"`\n    WebhookStatus enums.WebhookStatusEnum `json:\"webhook_status\" validate:\"required\"`\n}"
'''
